src/operator/nn/mkldnn/mkldnn_adaptive_pooling.cc

```diff
@@ -0,0 +1,97 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * Copyright (c) 2021 by Contributors
+ * \file mkldnn_adaptive_pooling.cc
+*/
+
+#if MXNET_USE_MKLDNN == 1
+
+#include "./mkldnn_adaptive_pooling-inl.h"
+
+namespace mxnet {
+namespace op {
+void MKLDNNAdaptivePoolingFwd::Init(
```

Comment:
Functions: MKLDNNAdaptivePoolingFwd::Init() and MKLDNNAdaptivePoolingFwd::Execute() are more or less the same as the one that are in the mkldnn_poolling.cc file (MKLDNNPoolingFwd::Init() && MKLDNNPoolingFwd::Execute()). What do you think about combining them together? So that the code won't be duplicated.

Reply:
Personally, I think I'm fine with it.  I will propose a fix in the next patch set.